packages/react-components/react-checkbox/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -95,4 +95,6 @@ export interface CheckboxOnChangeData {
  * State used in rendering Checkbox
  */
 export type CheckboxState = ComponentState<CheckboxSlots> &
-  Required<Pick<CheckboxProps, 'checked' | 'labelPosition' | 'shape' | 'size'>>;
+  Required<Pick<CheckboxProps, 'checked' | 'labelPosition' | 'shape' | 'size'>> & {
+    disabled: boolean | undefined;
```

Comment:
You can potentially require + pick `disabled` from `CheckboxProps` here and just do `disabled = false` in line 32 of `useCheckbox_unstable`.

```suggestion
  Required<Pick<CheckboxProps, 'checked' | 'disabled' | 'labelPosition' | 'shape' | 'size'>> & {
```

Reply:
My goal was to avoid adding `disabled={false}` to the underlying `input` in the default case. But after further investigation, I realized that the local `disabled` value in `useCheckbox_unstable` doesn't even get passed to the `input` (disabled gets passed via `nativeProps.primary` instead). Also, it looks like React handles `disabled={false}` by just not setting disabled on the control.

In short: sounds good, I'll make that change :)